pkg/jobs/jobs.go

```diff
@@ -121,8 +121,71 @@ type (
 		Timeout      time.Duration `json:"timeout"`
 		RetryDelay   time.Duration `json:"retry_delay"`
 	}
+
+	// Scheduler interface is used to represent a scheduler that is responsible
+	// to listen respond to triggers jobs requests and send them to the broker.
+	Scheduler interface {
+		Start(broker Broker) error
+		Add(trigger Trigger) error
+		Get(id string) (Trigger, error)
+		Delete(id string) error
+		GetAll() ([]Trigger, error)
+	}
+
+	// Trigger interface is used to represent a trigger.
+	Trigger interface {
+		Type() string
+		Infos() *TriggerInfos
+		// Schedule should return a channel on which the trigger can send job
+		// requests when it decides to.
+		Schedule() <-chan *JobRequest
+		// Unschedule should be used to clean the trigger states and should close
+		// the returns jobs channel.
+		Unschedule()
+	}
+
+	// TriggerStorage interface is used to represent a persistent layer on which
+	// triggers are stored.
+	TriggerStorage interface {
+		GetAll() ([]Trigger, error)
```

Comment:
[Optim] maybe we can add a `GetInfos()` method to return a list of TriggerInfos. It would more efficient than to return a list of triggers for the listing of the triggers in the web API, as it avoids to allocate a channel for each trigger.